src/freenet/clients/http/wizardsteps/BROWSER_WARNING.java

```diff
@@ -53,25 +56,17 @@ public void getStep(HTTPRequest request, PageHelper helper) {
 
 		HTMLNode contentNode = helper.getPageContent(WizardL10n.l10n("browserWarningPageTitle"));
 
-		String infoBoxHeader;
-		if(incognito) {
-			infoBoxHeader = WizardL10n.l10n("browserWarningIncognitoShort");
-		} else if (isRelativelySafe) {
-			infoBoxHeader = WizardL10n.l10n("browserWarningShortRelativelySafe");
-		} else {
-			infoBoxHeader = WizardL10n.l10n("browserWarningShort");
-		}
-		
+		String infoBoxHeader = infoBoxHeaderText(incognito, isRelativelySafe);
 		HTMLNode infoboxContent = helper.getInfobox("infobox-normal", infoBoxHeader, contentNode, null, false);
 
-		if(isOldFirefox) {
+		List<String> oldBrowserWarnings = oldBrowserWarnings(
+				incognito,
+				isOldFirefox,
+				showTabWarning);
+		if(!oldBrowserWarnings.isEmpty()) {
 			HTMLNode p = infoboxContent.addChild("p");
-			p.addChild("#", WizardL10n.l10n("browserWarningOldFirefox"));
-			if (showTabWarning) {
-				p.addChild("#", " " + WizardL10n.l10n("browserWarningFirefoxMightHaveClobberedTabs"));
-			} else if(!incognito) {
-				p.addChild("#", " " + WizardL10n.l10n("browserWarningOldFirefoxNewerHasPrivacyMode"));
-			}
+			p.addChild("#", oldBrowserWarnings.remove(0));
+			oldBrowserWarnings.forEach(s -> p.addChild("#", " " + s));
```

Comment:
This kind of obscurs what’s happening here.
```suggestion
			p.addChild("#", oldBrowserWarnings.stream().collect(Collectors.joining(" ")));
```
(Also, add a static import for the `joining` to keep the line shorter.)